Sources/SegmentAmplitude/AmplitudeSession.swift

```diff
@@ -40,128 +40,247 @@ public class AmplitudeSession: EventPlugin, iOSLifecycle {
+    @Atomic var sessionID: Int64 {
+        didSet {
+            storage.write(key: Storage.Constants.previousSessionID, value: sessionID)
+            //print("sessionID = \(sessionID)")
+        }
+    }
+    
+    @Atomic var lastEventTime: Int64 {
+        didSet {
+            storage.write(key: Storage.Constants.lastEventTime, value: lastEventTime)
         }
     }
     
+    public init() {
+        self.sessionID = storage.read(key: Storage.Constants.previousSessionID) ?? -1
+        self.lastEventTime = storage.read(key: Storage.Constants.lastEventTime) ?? -1
+        //print("startup sessionID = \(sessionID)")
+    }
+    
     public func update(settings: Settings, type: UpdateType) {
+        if type != .initial { return }
+            
         if settings.hasIntegrationSettings(key: key) {
             active = true
         } else {
             active = false
         }
+        
+        if sessionID == -1 {
+            startNewSession()
+        } else {
+            startNewSessionIfNecessary()
+        }
     }
     
     public func execute<T: RawEvent>(event: T?) -> T? {
-        if !active {
-            return event
-        }
+        guard let event else { return nil }
+        guard let event = defaultEventHandler(event: event) else { return nil }
         
-        var result: T? = event
-        switch result {
-        case let r as IdentifyEvent:
-            result = self.identify(event: r) as? T
-            lastEventFiredTime = Date()
-        case let r as TrackEvent:
-            result = self.track(event: r) as? T
-        case let r as ScreenEvent:
-            result = self.screen(event: r) as? T
-            lastEventFiredTime = Date()
-        case let r as AliasEvent:
-            result = self.alias(event: r) as? T
-            lastEventFiredTime = Date()
-        case let r as GroupEvent:
-            result = self.group(event: r) as? T
-            lastEventFiredTime = Date()
-        default:
-            break
+        if var trackEvent = event as? TrackEvent {
+            let eventName = trackEvent.event
+            
+            // check if time has elapsed and kick of a new session if it has.
+            // this will send events back through to do the tasks; nothing really happens inline.
+            startNewSessionIfNecessary()
+            
+            // if it's a start event, set a new sessionID
+            if eventName == Constants.ampSessionStartEvent {
+                sessionID = newTimestamp()
+            }
+            
+            // if it's amp specific stuff, disable all the integrations except for amp.
+            if eventName.contains(Constants.ampPrefix) || eventName == Constants.ampSessionStartEvent || eventName == Constants.ampSessionEndEvent {
+                var integrations = disableAllIntegrations(integrations: trackEvent.integrations)
+                integrations?.setValue(["session_id": sessionID], forKeyPath: KeyPath(key))
```

Comment:
Should the `session_id` be set outside this block for all events being fired? Currently I see that on Amplitude dashboard all my events are not being grouped together because the session iD is being passed as -1. In earlier version, the timestamp was being set to the session Id and it was correctly grouping all related session events.

<img width="1096" alt="Screenshot 2024-08-18 at 8 54 58 PM" src="https://github.com/user-attachments/assets/ffad8bf7-14cf-4445-aa20-5b773f2785ab">

<img width="1128" alt="Screenshot 2024-08-18 at 8 55 36 PM" src="https://github.com/user-attachments/assets/490541f0-cd37-414a-8427-27a802dcaf21">